jukebox/src/main/webapp/js/dbScript.js

```diff
@@ -0,0 +1,77 @@
+const NO_VIDEO = -1;
+
+var db = firebase.firestore();
+
+//Adds a song to the queue
+function addSongDB(song_ID, title, room_ID) {
+    db.collection('songs').where("SONG_ID", "==", song_ID)
+    .get()
+    .then(function(querySnapshot) {
+        if(querySnapshot.exists){ //Song is already in database, just add to queue
+            db.collection("queue").doc(song_ID).set( {
+                SONG_ID: song_ID,
+                ROOM_ID: room_ID,
+                DATE_ADDED: Date.now()
+            });
+        } else {
+            db.collection("songs").doc(song_ID).set( {
+                SONG_ID: song_ID,
+                SONG_NAME: title
+            });
+            db.collection("queue").doc(song_ID).set( {
```

Comment:
Since both the songs collection and the queue collection share every attribute except SONG_NAME, maybe we could merge them to form a single queue collection. For example, if we just added SONG_NAME as a field in the queue collection, then we would only need the queue collection (unless there's some other functionality of the 'songs' collection I'm missing).

Reply:
I believe the its just an abstraction due to a many to many relationship. Multiple songs could belong in multiple queues (having names and date added being variable factors). However, the amount of space of having overlapping songs being added to the database (such as the same song being added to two different queues) would probably be minimal and wouldn't make a big difference if we just added songs straight to the queue collection.

Reply:
Ah, yeah I didn't think about the space being saved by storing the songs.